lib/include/event-driven/vIPT.h

```diff
@@ -50,6 +51,8 @@ class vIPT {
 
     vIPT();
 
+
+    const cv::Mat getQ();
```

Comment:
const cv::Mat& getQ();

you wanted to return a reference to the Mat right?

Reply:
I changed it so that the calling function has a choice to get a copy or direct reference of the matrix.

Reply:
How do you mean? Here the only option is to get a copy of Q. 
if you make the function `const cv::Mat & getQ()`;
and they call it with 
`cv::Mat q = getQ();` they get a copy
`const cv::Mat &q = getQ();` they get the reference
right?

Reply:
With the definition `const cv::Mat getQ();`, can't we also pass a reference by calling like this:
`const cv::Mat &q = getQ();` ?

Reply:
My first guess is that what you want to do wouldn't compile (but I guess you have tested it and it does). So then my guess would be as you say, the function returns a copy and you are just making it a reference to it. It probably compiles to the same thing. always a copy.

Have you ever used a `yarp::os::BufferedPort<Bottle>` to write some data and used the `Bottle & prepare()` function. If you forget to use the reference when you are calling `prepare()` (like `Bottle b = port.prepare()` you get the copy of the object, which you modify with your content, but when you call `BufferedPort::write()` it writes the original `Bottle` that wasn't modified? That is the same functionality that (in this case) we want.

Reply:
I tested and inspected the memory addresses in both cases. It behaves as suspected -- it returns a copy and then references that copy. So, it does not work for calls that need to use references to modify variables (like in `BufferedPort`). This is not a problem in our case since a const is returned. However, this is inefficient because of extra copying all the time. I'll change it back so that `getQ()` returns a reference.

Reply:
pushed new commit